app/src/androidTest/java/org/akvo/flow/activity/FreeTextSurveyTest.java

```diff
@@ -0,0 +1,107 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.akvo.flow.activity;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.contrib.DrawerActions;
+import android.support.test.rule.ActivityTestRule;
+
+import org.akvo.flow.R;
+import org.akvo.flow.domain.Survey;
+import org.akvo.flow.testhelper.SurveyInstaller;
+import org.akvo.flow.util.FileUtil;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
+import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.akvo.flow.tests.R.raw.freetextsurvey;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.not;
+
+public class FreeTextSurveyTest {
+
+    @Rule
+    public ActivityTestRule<SurveyActivity> rule = new ActivityTestRule<>(SurveyActivity.class);
+
+    private Context runtimeContext;
+    private SurveyInstaller installer;
+
+    @Before
+    public void init()
+    {
+        runtimeContext = InstrumentationRegistry.getContext();
+        installer      = new SurveyInstaller(rule.getActivity().getApplicationContext());
+    }
+
+    @Test
+    public void canFillFreeTextQuestion() throws IOException {
+        fillFreeTextQuestion(freetextsurvey, "This is an answer to your question");
+        onView(allOf(withClassName(endsWith("Button")), withText(R.string.submitbutton))).check(matches((isEnabled())));
+    }
+
+    @Test
+    public void ensureCantSubmitEmptyFreeText() throws IOException {
+        fillFreeTextQuestion(freetextsurvey, "");
+        onView(allOf(withClassName(endsWith("Button")), withText(R.string.submitbutton))).check(matches(not(isEnabled())));
+    }
+
+    private Survey fillFreeTextQuestion(int surveyResId, String text) throws IOException {
+        Survey survey = getSurvey(surveyResId);
+
+        openDrawer();
+        //Check for specified survey name, make sure it matches what's displayed and click it
```

Comment:
lines 83-92: I would say these comments are not necessary as it seems obvious what the code does. When it is not obvious what the code does it is better to refactor it and name the method in such a way that is self explanatory. See for example this blog https://blog.codinghorror.com/coding-without-comments/.